application/apps/indexer/processor/src/search/searchers/values.rs

```diff
@@ -44,6 +59,15 @@ pub type ValueSearchHolder = BaseSearcher<ValueSearchState>;
 
 impl ValueSearchHolder {
     pub fn setup(&mut self, terms: Vec<String>) -> Result<(), SearchError> {
+        let invalid = terms
+            .iter()
+            .filter(|f| !is_valid(f))
+            .cloned()
+            .collect::<Vec<String>>()
+            .join("; ");
+        if !invalid.is_empty() {
+            Err(SearchError::Input(format!("Invalid filters: {invalid}")))?;
+        }
```

Comment:
you are using basically the same algorithm twice...why not move it to the search module?
BTW this version of the filtering is nicer than converting to None/Some

Reply:
do you mean `processor/src/search/searchers/mod.rs`?

but we are checking different things: in case of regular search we are checking SearchFilter in case of values - String ... check would be different in any way  :/

also then we get holder from state, we get some of holders: `get_search_holder` or `get_search_values_holder`. So in any way we are "working" with target holder